constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/ValueAt.java

```diff
@@ -0,0 +1,66 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.DiscreteProfile;
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.LinearProfile;
+import gov.nasa.jpl.aerie.constraints.model.Profile;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.IntervalMap;
+import gov.nasa.jpl.aerie.constraints.time.Segment;
+import gov.nasa.jpl.aerie.constraints.time.Spans;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
+
+import java.util.Set;
+
+public record ValueAt(
+    Expression<DiscreteProfile> profile,
+    Expression<Spans> timepoint
+) implements Expression<DiscreteProfile> {
+
+
+  @Override
+  public DiscreteProfile evaluate(
+      final SimulationResults results,
+      final Interval bounds,
+      final EvaluationEnvironment environment)
+  {
+    final var res = this.profile.evaluate(results, bounds, environment);
+    final var time = timepoint.evaluate(results, bounds, environment);
+    final var timepoint = time.iterator().next().interval().start;
+    //REVIEW: SHOULD ASSERT A BUNCH OF THINGS HERE SO IT IS NOT WRONGLY USED
+    valueAt(res, timepoint);
+    return new DiscreteProfile(IntervalMap.<SerializedValue>builder()
+                                          .set(Segment.of(bounds, valueAt(res, timepoint)))
+                                          .build());
+  }
+
+
+  public static SerializedValue valueAt(Profile<?> profile, Duration time){
+    if(profile instanceof DiscreteProfile discreteProfile){
+      return discreteProfile.valueAt(time);
+    } else if (profile instanceof LinearProfile linearProfile){
+      return linearProfile.valueAt(time);
+    } else{
+      throw new Error("Profile is neither discrete or linear");
+    }
```

Comment:
I think this is unnecessary, since the Profile interface already has the valueAt function. In `evaluate` you can just call `res.valueAt(timepoint)`.